plugins/events/create.js

```diff
@@ -78,12 +78,30 @@ module.exports = () => ({
                     userFactory.get({ username, scmContext })
                 ]).then(([pipeline, user]) => user.getPermissions(pipeline.scmUri)
                     // Check if user has push access
+                    // eslint-disable-next-line consistent-return
                     .then((permissions) => {
                         if (!permissions.push) {
-                            throw boom.unauthorized(`User ${username} `
-                              + 'does not have push permission for this repo');
+                            const newAdmins = pipeline.admins;
+
+                            delete newAdmins[username];
+                            // This is needed to make admins dirty and update db
+                            pipeline.admins = newAdmins;
+
+                            return pipeline.update()
+                                .then(() => {
+                                    throw boom.unauthorized(`User ${username} `
+                                    + 'does not have push permission for this repo');
+                                });
                         }
                     })
+                    // user has good permissions, add the user as an admin
+                    .then(() => {
+                        const newAdmins = pipeline.admins;
+
+                        newAdmins[username] = true;
+                        // This is needed to make admins dirty and update db
```

Comment:
still remain update db in the comment.

Reply:
Yes, we need to make it dirty to update db.
https://github.com/screwdriver-cd/models/blob/master/lib/base.js#L30

Reply:
I think you should change the comment `and update db` -> `to update db`.